AstSemantics.md

```diff
@@ -111,6 +112,9 @@ Integer loads can specify a *storage size* which is smaller than the result type
 well as a signedness which determines whether the bytes are sign- or zero-
 extended into the result type.
 
+Each `load` and `store` specifies the [linear memory index](Modules.md#index-spaces)
+to operate on with an unsigned integer immediate.
+
```

Comment:
I don't understand, how can that work? A C or C++ pointer can point to either the stack or the heap, as @lukewagner and I said, so they can't be in separate address spaces, can they?

Are you suggesting fat pointers? Or some sophisticated type-based analysis like Cheerp does? Or am I missing something obvious? :)